### PR TITLE
FRIN23-315 navigation targets

### PR DESCRIPTION
**Pull request by AlanCornthwaiteKatalyst**

### CODE REVIEW

File: app/helpers/katalyst/navigation/frontend/builder.rb [hunk 1]
```diff
@@ -46,11 +46,11 @@ def render_heading(heading)
         end
 
         def render_link(link)
-          link_to(link.title, link.url)
+          link_to(link.title, link.url, **link.item_options)
         end
 
         def render_button(link)
-          link_to(link.title, link.url, method: link.http_method)
+          link_to(link.title, link.url, **link.item_options)
```
Comment (sobakasu) [inline app/helpers/katalyst/navigation/frontend/builder.rb:53]: the ** is unnecessary

**Review by sobakasu — APPROVED** [review]
looks ok some optional amendments

File: app/models/katalyst/navigation/item.rb [hunk 2]
```diff
@@ -24,6 +28,22 @@ def layout?
...
+      def options_for_target
+        return {} if target == "self"
+
+        options = { target: target }
+
+        unless target == "_blank" || target == "_top"
+          options = { data: { turbo: true, turbo_frame: target } }
+        end
+        options
+      end
```
Comment (sobakasu) [inline app/models/katalyst/navigation/item.rb:45]: (opt) case statement might be more readable

File: app/models/katalyst/navigation/button.rb [hunk 3]
```diff
@@ -14,6 +14,17 @@ class Button < Item
...
+      def options_for_target
+        options = super
+
+        if target == "_blank" || target == "_top" || target == "self"
+          options.deep_merge!({ data: { method: http_method } })
+        else
+          options.deep_merge!({ data: { turbo_method: http_method } })
+        end
+        options
+      end
```
Comment (sobakasu) [inline app/models/katalyst/navigation/button.rb:27]: (opt) case statement might be more readable

File: app/helpers/katalyst/navigation/frontend/builder.rb (hunk 1 above)
Comment (AlanCornthwaiteKatalyst) [reply app/helpers/katalyst/navigation/frontend/builder.rb]: fixed

File: app/models/katalyst/navigation/item.rb (hunk 2 above)
Comment (AlanCornthwaiteKatalyst) [reply app/models/katalyst/navigation/item.rb:45]: if we were to add new targets (more turbo frames etc) we wont have to update this code
and have the special cases `_top` etc mapped out

**Pull request merged**